app/app.py

```diff
@@ -253,22 +254,26 @@ def create_event():
     return render_template('createEvent.html', form=form, flair=form.flair.choices)
 
 
-#MyProfile
-@app.route('/viewProfile/<username>', methods=['GET', 'POST'])
+#View Someone's Profile
+#@app.route('/viewProfile/<username>', methods=['GET', 'POST'])
+#@login_required
+#def viewProfile(username):
+#    user = User.query.get(username)
+#    return redirect(url_for('viewProfile', username=username))
+
+
+#Edit My Profile
+@app.route('/editProfile/<username>', methods=['GET', 'POST'])
 @login_required
-def viewAccount(username):
-    user = user.query.get(username)
+def editProfile(username):
+    user = User.query.get(username)
```

Comment:
query.get() only searches the Primary Key of the table. It only works with a primary key integrer. Try this instead. If you want the username to be in the url than:

you need to use : user = User.query.filter_by(username=username).first()

or use primary key value and use:

`@app.route('/editProfile/<int:userID>, methods=['GET', 'POST'])
@login_required
def editProfile(userID):
    user = User.query.get(userID)`